### PR TITLE
Pywsidi merge

### PR DESCRIPTION
**Pull request by alex-virodov**
Initial PR for merging https://github.com/codybum/pywsidi into this repo. Not tested, just a sketch to check complexity. Will do extensive testing if the direction is correct.

### CODE REVIEW

**Comment by huvers** [comment]
Thanks @alex-virodov.  @anishag14 can you please review?

File: README.md [hunk 1]
```diff
@@ -19,18 +19,33 @@ Descriptions of de-id tools to go here
...
 To use the tool
 
+    cd synthetic_phi/ultrasound/src
 	python generate_synth_phi.py -i input_dir -o output_dir -m num_synth_dicoms
 
+# whole slide image synthetic_phi
+
+This library allows python applications to deidentify WSI
+
+* iSyntax: deident_isyntax_file(ident_file_path, deident_file_path)
```
Comment (anishag14) [inline README.md:41]: What is iSyntax and SVS?

File: synthetic_phi/whole_slide_image/src/isyntax.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
+import os
+import traceback
+
+from lxml import etree as ET
+import base64
+import io
+from PIL import Image, ImageDraw
```
Comment (anishag14) [inline synthetic_phi/whole_slide_image/src/isyntax.py:6]: Can you run isort on your code to sort this import list?

File: synthetic_phi/whole_slide_image/src/isyntax.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    xml_declaration = '<?xml version="1.0" encoding="UTF-8"?>\n'
+    search_string = 'PMSVR="IString"/><Attribute Name="PIIM_DP_SCANNER_CALIBRATION_STATUS"'
+    replace_string = 'PMSVR="IString"></Attribute><Attribute Name="PIIM_DP_SCANNER_CALIBRATION_STATUS"'
+
+    output = xml_declaration + ET.tostring(root, encoding='utf-8', method='xml', pretty_print=True).decode('utf-8')
+    output = output.replace(search_string, replace_string)
+    return output.encode('utf-8')
+
+def getisyntaxheader(file_path):
```
Comment (anishag14) [inline synthetic_phi/whole_slide_image/src/isyntax.py]: Can you rename this as get_isyntax_header similar fashion to other funtions?

Comment (anishag14) [inline synthetic_phi/whole_slide_image/src/isyntax.py]: Can you rename this as get_isyntax_header similar fashion to other funtions?

File: synthetic_phi/whole_slide_image/src/isyntax.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    for index in range(0, len(a_string), n):
+        split_strings.append(a_string[index: index + n])
+
+    payload = ''
+    for str in split_strings:
+        payload += str + '\n'
+
+    return payload[:-1]
+
+def parseXMLheader(header_data, image_text1=None, image_text2=None):
```
Comment (anishag14) [inline synthetic_phi/whole_slide_image/src/isyntax.py]: Are you following coding style where local functions are using camel case? if not, can you rename above function name as parse_xml_header?

File: synthetic_phi/whole_slide_image/src/isyntax.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+    with io.BytesIO() as output:
+        img.save(output, format='JPEG')
+        contents = output.getvalue()
+        base64_bytes = base64.b64encode(contents)
+        base64_string = base64_bytes.decode("utf-8")
+
+    return base64_string
+
+def base64Split(a_string):
```
Comment (anishag14) [inline synthetic_phi/whole_slide_image/src/isyntax.py]: same here. Are you following coding style where local functions are using camel case? if not, can you rename above function name as base_64_split?

File: synthetic_phi/whole_slide_image/src/isyntax.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
+import os
+import traceback
+
+from lxml import etree as ET
+import base64
+import io
+from PIL import Image, ImageDraw
+
+def generate_image(x, y, image_text1=None, image_text2=None):
+    img = Image.new('RGB', (x, y), color=(73, 109, 137))
```
Comment (anishag14) [inline synthetic_phi/whole_slide_image/src/isyntax.py]: (73, 109, 137) are these random numbers or hold any significance? Can you make a constant with a name so we know what these numbers are for?

File: synthetic_phi/whole_slide_image/src/isyntax.py [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+    d = ImageDraw.Draw(img)
+    d.text((100, 100), "REMOVED DATA 1", fill=(255, 255, 0))
```
Comment (anishag14) [inline synthetic_phi/whole_slide_image/src/isyntax.py]: same here and below. You can do something like: https://github.com/project-medid/project-medid/blob/main/synthetic_phi/ultrasound/src/utils.py#L80

File: synthetic_phi/whole_slide_image/src/isyntax.py (hunk 6 above)
Comment (anishag14) [reply synthetic_phi/whole_slide_image/src/isyntax.py]: an example: https://github.com/project-medid/project-medid/blob/main/synthetic_phi/ultrasound/src/utils.py#L80

File: synthetic_phi/whole_slide_image/src/isyntax.py (hunk 4 above)
Comment (anishag14) [reply synthetic_phi/whole_slide_image/src/isyntax.py]: Also, what is image_text1 and 2?

**Review by anishag14 — COMMENTED** [review]
Hi Cody,
You have added the folder under syncthetic-phi/ named whole_slide_image/ . I think it should be the named after a modality like pathology. I open to suggestions, so please let me know what you think.

File: README.md (hunk 1 above)
Comment (alex-virodov) [reply README.md:41]: Those are whole-slide image formats, similar conceptually to DICOM (though significantly different in details).

File: synthetic_phi/whole_slide_image/src/isyntax.py (hunk 2 above)
Comment (alex-virodov) [reply synthetic_phi/whole_slide_image/src/isyntax.py:6]: Done.

File: synthetic_phi/whole_slide_image/src/isyntax.py (hunk 6 above)
Comment (alex-virodov) [reply synthetic_phi/whole_slide_image/src/isyntax.py]: Good idea, thank you. No special significance. I added a named constant DEFAULT_BACKGROUND_BLUE.

File: README.md (hunk 1 above)
Comment (alex-virodov) [reply README.md:41]: I couldn't find a single good reference for whole slide image formats, so I'm leaving it as-is.

File: synthetic_phi/whole_slide_image/src/isyntax.py (hunk 7 above)
Comment (alex-virodov) [reply synthetic_phi/whole_slide_image/src/isyntax.py]: Done.

**Comment by alex-virodov** [comment]
Also, I tested the code this time, so this is ready for proper merge after review is complete.

File: synthetic_phi/whole_slide_image/src/isyntax.py (hunk 3 above)
Comment (alex-virodov) [reply synthetic_phi/whole_slide_image/src/isyntax.py]: Done.

Comment (alex-virodov) [reply synthetic_phi/whole_slide_image/src/isyntax.py]: Done.

File: synthetic_phi/whole_slide_image/src/isyntax.py (hunk 4 above)
Comment (alex-virodov) [reply synthetic_phi/whole_slide_image/src/isyntax.py]: Done. image_text1 and image_text2 are the fake text to put in a fake image that replaces the identified image. They were unused (and still are), but I'm passing them a bit deeper now.

File: synthetic_phi/whole_slide_image/src/isyntax.py (hunk 5 above)
Comment (alex-virodov) [reply synthetic_phi/whole_slide_image/src/isyntax.py]: Done.

**Review by huvers — APPROVED** [review]

**Comment by alex-virodov** [comment]
Note: the force pushes were to sign the commits, no content was changed.

File: README.md (hunk 1 above)
Comment (anishag14) [reply README.md:41]: Can we add what you told me that they are they are two different formats for this modality?

**Pull request merged**